packages/components/rating/src/Rating.tsx

```diff
@@ -1 +1,182 @@
-export const Rating = () => <>rating</>
+import { cx } from 'class-variance-authority'
+import {
+  type ChangeEvent,
+  ComponentPropsWithoutRef,
+  forwardRef,
+  type MouseEvent,
+  PropsWithChildren,
+  useCallback,
+  useRef,
+  useState,
+} from 'react'
+
+import { RatingStar } from './RatingStar'
+import { type Size } from './types'
+import { getNearestDecimal, getStarValue, splitAt } from './utils'
+
+export interface RatingProps extends PropsWithChildren<ComponentPropsWithoutRef<'div'>> {
+  /**
+   * Use the `defaultValue` prop to set the default value of the input, on a from 0 to 5.
+   *
+   * Use this when you want to use it in an uncontrolled manner
+   */
+  defaultValue?: number
+  /**
+   * The value is the number of the rating selected, on a scale from 0 to 5.
+   *
+   * Use this when you want to use it in a controlled manner,
+   * in conjunction with the `onValueChange` prop
+   */
+  value?: number
+  /**
+   * Event handler called when the value changes.
+   */
+  onValueChange?: (value: number) => void
+  /**
+   * Sets the component as interactive or not.
+   * @default undefined
+   */
+  readOnly?: boolean
+  /**
+   * When `true`, prevents the user from interacting.
+   * @default false
+   */
+  disabled?: boolean
+  /**
+   * Sets the size of the stars.
+   * @default 'md'
+   */
+  size?: Size
```

Comment:
Why not doing as usual, and using the `RatingStarstylesProps` from styles? Just being curious... :sweat_smile:

Reply:
Mostly because I found it more legible to declare the type this way 😄

Reply:
Got it, but we usually type (afaik) with `VariantProps` from cva in such cases... Nothing blocking for me, maybe someone else would have his 2cents to share with us? :slightly_smiling_face: